mcs/class/System/System.Net/HttpListenerRequest.cs

```diff
@@ -403,7 +403,7 @@ internal bool FlushInput ()
 		}
 
 		public bool IsLocal {
-			get { return IPAddress.IsLoopback (RemoteEndPoint.Address); }
+			get { return LocalEndPoint.Address.Equals (RemoteEndPoint.Address); }
```

Comment:
So in a windows virtual machine with 2 bridge adapters to the same network. When we have a HttpListenerRequest with LocalEndPoint equal to 192.168.1.103:9000 and RemoteEndPoint equal to 192.168.1.104:9000 HttpListenerRequest.IsLocal returns false.